src/Form/KeyForm.php

```diff
@@ -111,6 +111,23 @@ public function submitForm(array &$form, FormStateInterface $form_state) {
   /**
    * {@inheritdoc}
    */
+  public function validateForm(array &$form, FormStateInterface $form_state) {
+    // Only run key settings validation if the form is being submitted
+    if($form_state->isSubmitted()) {
+      $plugin_settings = (new FormState())->setValues($form_state->getValue('key_settings'));
+      $plugin = $this->manager->createInstance($form_state->getValue('key_type'), []);
+      $plugin->validateConfigurationForm($form, $plugin_settings);
+      // Reinject errors from $plugin_settings into $form_state
+      foreach ($plugin_settings->getErrors() as $field => $error) {
```

Comment:
I think this is fine. I snooped around the FormStateInterface to see if there was a merge function or something close and didn't see it. This isn't a very common use case, but we may want to look at something like blocks or Views, which have settings per plugin. I wrote Password Policy to use a CTools wizard, but that had independent forms within a modal window.